lib/worker.go

```diff
@@ -312,28 +368,31 @@ func (m *masterClient) HandleHeartbeat(msg *HeartbeatPongMessage) {
 	m.mu.Lock()
 	defer m.mu.Unlock()
 
-	// TODO think about whether to distinct msg.Epoch > m.masterEpoch
-	// and msg.Epoch < m.masterEpoch
-	if msg.Epoch != m.masterEpoch {
-		log.L().Info("epoch does not match",
+	if msg.Epoch < m.masterEpoch {
+		log.L().Info("epoch does not match, ignore stale heartbeat",
 			zap.Any("msg", msg),
 			zap.Int64("master-epoch", m.masterEpoch))
 		return
 	}
+
+	if msg.Epoch > m.masterEpoch {
+		m.masterEpoch = msg.Epoch
+	}
```

Comment:
Should `msg.Epoch == m.masterEpoch` be never happened.

Reply:
This should be the norm.